errorlint/lint.go

```diff
@@ -188,10 +191,46 @@ func LintErrorComparisons(info *TypesInfoExt) []analysis.Diagnostic {
 			continue
 		}
 
-		lints = append(lints, analysis.Diagnostic{
+		diagnostic := analysis.Diagnostic{
 			Message: fmt.Sprintf("comparing with %s will fail on wrapped errors. Use errors.Is to check for a specific error", binExpr.Op),
 			Pos:     binExpr.Pos(),
-		})
+		}
+
+		// Add suggested fix
+		var errVar, targetErr ast.Expr
+		// Identify which side is the error variable and which is the sentinel error
+		if isErrorType(info.TypesInfo, binExpr.X) && isErrorType(info.TypesInfo, binExpr.Y) {
+			// If both sides are errors, we'll use them as is
+			errVar = binExpr.X
+			targetErr = binExpr.Y
+		} else if isErrorType(info.TypesInfo, binExpr.X) {
+			errVar = binExpr.X
+			targetErr = binExpr.Y
+		} else {
+			errVar = binExpr.Y
+			targetErr = binExpr.X
+		}
+
+		negated := binExpr.Op == token.NEQ
+
+		// Build the suggested fix - preserve the original order of parameters
+		var replacement string
+		if negated {
+			replacement = fmt.Sprintf("!errors.Is(%s, %s)", exprToString(errVar), exprToString(targetErr))
+		} else {
+			replacement = fmt.Sprintf("errors.Is(%s, %s)", exprToString(errVar), exprToString(targetErr))
+		}
```

Comment:
The code could be simplified I think

```suggestion
		// Build the suggested fix - preserve the original order of parameters
		replacement := fmt.Sprintf("errors.Is(%s, %s)", exprToString(errVar), exprToString(targetErr))
		if negated {
			replacement = "!" + replacement
		}
```

But this version might be cleaner

```suggestion
		// Build the suggested fix - preserve the original order of parameters
		msgPattern := "errors.Is(%s, %s)"
		if negated {
			msgPattern = "!errors.Is(%s, %s)"
		}
		replacement := fmt.Sprintf(msgPattern, exprToString(errVar), exprToString(targetErr))
```

Reply:
I prefer the first suggestion, the one that only prepends the `!` if needed